lib/x/errors/too_many_requests.rb

```diff
@@ -1,23 +1,31 @@
 require_relative "client_error"
+require_relative "../rate_limit"
 
 module X
   class TooManyRequests < ClientError
-    def limit
-      response["x-rate-limit-limit"].to_i
+    def rate_limit
+      rate_limits.values.max_by(&:reset_at)
     end
 
-    def remaining
-      response["x-rate-limit-remaining"].to_i
+    def rate_limits
+      @rate_limits ||= limit_types.to_h { |type| [type, RateLimit.new(type: type, response: response)] }
+        .select { |_, limit| limit.remaining.zero? }
```

Comment:
Rubocop and ruby standard aren't agreed here.

Rubocop gives : 

```
lib/x/errors/too_many_requests.rb:12:9: C: [Correctable] Layout/MultilineMethodCallIndentation: Align .select with .to_h on line 11.
        .select { |_, limit| limit.remaining.zero? }
        ^^^^^^^
```

If I correct, standard gives : 

```
standard: Use Ruby Standard Style (https://github.com/standardrb/standard)
  lib/x/errors/too_many_requests.rb:12:35: Layout/MultilineMethodCallIndentation: Use 2 (not 28) spaces for indenting an expression in an
 assignment spanning multiple lines.
```

I prefer the "standard" version

Reply:
When Standard and RuboCop defaults conflict, I always go with Standard Ruby. Feel free to edit `.rubocop.yml` to make it consistent with Standard Ruby, so the linter check passes.